analysis/priestdiscipline/src/modules/shadowlands/legendaries/ClarityOfMindEvang.tsx

```diff
@@ -0,0 +1,140 @@
+import SPELLS from 'common/SPELLS';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import ItemManaGained from 'parser/ui/ItemManaGained';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+import { POWER_WORD_SHIELD_ATONEMENT_DUR } from '@wowanalyzer/priest-discipline/src/constants';
+
+import AtonementAnalyzer, { AtonementAnalyzerEvent } from '../../core/AtonementAnalyzer';
+
+const CLARITY_EXTENSION_DURATION = 6000;
+const EVANG_EXTENSION_DURATION = 6000;
+const POWER_WORD_SHIELD_MANA_COST = 1550;
+
+class ClarityOfMindEvang extends Analyzer {
+  private atonementHealing: number = 0;
+  raptureBuffActive = false;
+  raptureShields = [] as any;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active =
+      this.selectedCombatant.hasLegendary(SPELLS.CLARITY_OF_MIND) &&
+      this.selectedCombatant.hasTalent(SPELLS.EVANGELISM_TALENT);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(AtonementAnalyzer.atonementEventFilter, this.handleAtone);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER), this.checkRapture);
```

Comment:
```suggestion
    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell([SPELLS.POWER_WORD_SHIELD, SPELLS.RAPTURE]), this.checkRapture);
```

add `.spell([SPELLS.POWER_WORD_SHIELD, SPELLS.RAPTURE])` to only trigger this event handler on those spells